### PR TITLE
fix for checkbox opacity issues in safari

### PR DESCRIPTION
**Pull request by sendilkumarn**
#557 

 Variables - changed cubic-beizer for linear-out-slow-in based on [this](https://github.com/google/material-design-lite/blob/master/src/_variables.scss)

Calculation of opacity value in safari has some issues when rendering and it corrects itself upon moving focus out of window. i.e., on recalculating.

moved background and opacity from `.md-checkbox-background` and set them dynamically upon checkbox transition.

### CODE REVIEW

File: src/core/style/_variables.scss [hunk 1]
```diff
@@ -24,7 +24,7 @@ $md-toggle-size: 20px !default;
 // TODO(jelbourn): all of these need to be revisited
 
 // The default animation curves used by material design.
-$md-linear-out-slow-in-timing-function: cubic-bezier(0.0, 0.0, 0.2, 0.1) !default;
```
Comment (jelbourn) [inline src/core/style/_variables.scss]: I'd prefer not to change this in this PR and instead do a review directly with the Material Design UX team.

File: src/components/checkbox/checkbox.scss [hunk 2]
```diff
@@ -359,6 +357,13 @@ md-checkbox {
   }
 }
 
+
+.md-checkbox-unchecked {
+  .md-checkbox-background {
+    background-color: none;
```
Comment (jelbourn) [inline src/components/checkbox/checkbox.scss]: I don't believe `none` is a valid value for `background-color`. Did you mean` transparent`?

File: src/core/style/_variables.scss (hunk 1 above)
Comment (sendilkumarn) [inline src/core/style/_variables.scss]: Can you guide on how?

File: src/components/checkbox/checkbox.scss (hunk 2 above)
Comment (sendilkumarn) [inline src/components/checkbox/checkbox.scss]: Yeah transperant.

File: src/core/style/_variables.scss (hunk 1 above)
Comment (jelbourn) [inline src/core/style/_variables.scss]: I mean to just revert this line here.

Comment (sendilkumarn) [inline src/core/style/_variables.scss]: Okay then. Will change this and background thing

**Comment by jelbourn** [comment]
LGTM

**Comment by jelbourn** [comment]
Thanks!

**Pull request merged**